src/test/java/org/mongojack/TestDBUpdateSerialization.java

```diff
@@ -153,6 +157,30 @@ public void testSimpleMap() {
         assertThat(coll.findOneById("id").map, equalTo(map));
     }
 
+    // Test to detect presence of issue https://github.com/devbliss/mongojack/issues/98
+    @Test
+    public void testUpdateOfNestedRepeatedAttributeName() {
+        coll2 = getCollection(NestedRepeatedAttributeName.class, String.class);
+
+        Calendar cal = Calendar.getInstance();
+        cal.set(0, 0, 0, 0, 0, 10);
+        Date d_1 = cal.getTime();
+        cal.set(0, 0, 0, 0, 0, 20);
+        Date d_2 = cal.getTime();
+    	
+        NestedRepeatedAttributeName nested = new NestedRepeatedAttributeName();
+        nested.inner.timestamp = d_1;
```

Comment:
`nested` is a pretty confusing name because it means essentially the same thing as `inner`, which isn't what you mean at all. How about if we just call it `obj` instead?